x-pack/plugins/monitoring_collection/server/lib/verify_mappings.ts

```diff
@@ -0,0 +1,95 @@
+/*
```

Comment:
This seems to work well with a custom test case of adding in a collector and ensuring the mappings don't match or are non existent. 

If we like this approach, I can provide unit tests for this file

Reply:
At what point would the mappings be added to support the new metrics?

Reply:
I'd expect the mapping change to happen at the same time so anyone adding metrics should be building ES from source (`yarn es source`) with changes to the mappings to ensure the fields will be properly mapped. The ES PR technically needs to merge first to make sure the mappings are up to date before the Kibana PR is merged